corporate/views/billing_page.py

```diff
@@ -159,77 +111,11 @@ def billing_home(
-                plan.status == CustomerPlan.SWITCH_TO_ANNUAL_AT_END_OF_CYCLE
-            )
-            licenses = last_ledger_entry.licenses
-            licenses_at_next_renewal = last_ledger_entry.licenses_at_next_renewal
-            seat_count = get_latest_seat_count(user.realm)
-
-            # Should do this in JavaScript, using the user's time zone
-            if plan.is_free_trial():
-                assert plan.next_invoice_date is not None
-                renewal_date = "{dt:%B} {dt.day}, {dt.year}".format(dt=plan.next_invoice_date)
-            else:
-                renewal_date = "{dt:%B} {dt.day}, {dt.year}".format(
-                    dt=start_of_next_billing_cycle(plan, now)
-                )
-
-            renewal_cents = renewal_amount(plan, now)
-            charge_automatically = plan.charge_automatically
-            assert customer.stripe_customer_id is not None  # for mypy
-            stripe_customer = stripe_get_customer(customer.stripe_customer_id)
-            if charge_automatically:
-                payment_method = payment_method_string(stripe_customer)
-            else:
-                payment_method = "Billed by invoice"
-
-            fixed_price = (
-                cents_to_dollar_string(plan.fixed_price) if plan.fixed_price is not None else None
-            )
-
-            billing_frequency = CustomerPlan.BILLING_SCHEDULES[plan.billing_schedule]
-
-            if plan.price_per_license is None:
-                price_per_license = ""
-            elif billing_frequency == "Annual":
-                price_per_license = format_money(plan.price_per_license / 12)
-            else:
-                price_per_license = format_money(plan.price_per_license)
-
-            context.update(
-                plan_name=plan.name,
-                has_active_plan=True,
-                free_trial=plan.is_free_trial(),
-                downgrade_at_end_of_cycle=downgrade_at_end_of_cycle,
-                automanage_licenses=plan.automanage_licenses,
-                switch_to_annual_at_end_of_cycle=switch_to_annual_at_end_of_cycle,
-                licenses=licenses,
-                licenses_at_next_renewal=licenses_at_next_renewal,
-                seat_count=seat_count,
-                renewal_date=renewal_date,
-                renewal_amount=cents_to_dollar_string(renewal_cents),
-                payment_method=payment_method,
-                charge_automatically=charge_automatically,
-                stripe_email=stripe_customer.email,
-                CustomerPlan=CustomerPlan,
-                onboarding=onboarding,
-                billing_frequency=billing_frequency,
-                fixed_price=fixed_price,
-                price_per_license=price_per_license,
-                success_message=success_message,
-            )
+    billing_session = RealmBillingSession(user=None, realm=user.realm)
+    main_context = billing_session.get_billing_page_context()
+    if main_context:
```

Comment:
>  billing_session = RealmBillingSession(user=None, realm=user.realm)

used `user=None` here to keep `acting_user=null` for `write_to_audit_log` (called via `make_end_of_cycle_updates_if_needed` in `get_billing_page_context`)

Should we instead add an optional argument `system_job` in `write_to_audit_log` instead of using `user=None` for `RealmBillingSession` for these cases.

Reply:
Hmm, yeah, I think the right path here probably involves having the `self.write_to_audit_log` call inside `make_end_of_cycle_updates_if_needed` include pass a parameter to override the `acting_user`; I suppose simplest would be just to pass an `system_job_acting_user=True` parameter to `write_to_audit_log` that forces `acting_user=None`.